### PR TITLE
chore: warn on shadowSupportMode 'any'

### PR DESCRIPTION
**Pull request by ekashida**
## Details

This avoids making the removal of `shadowSupportMode="any"` a breaking change.

## Does this pull request introduce a breaking change?

* ✅ No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

* ✅ No, it does not introduce an observable change.

## GUS work item

W-14974451

### CODE REVIEW

**Review by jye-sf — APPROVED** [review]

File: packages/@lwc/integration-karma/test/mixed-shadow-mode/LightningElement.shadowSupportMode/index.spec.js [hunk 1]
```diff
@@ -15,7 +16,14 @@ describe('shadowSupportMode static property', () => {
...
             createElement('x-valid', { is: Valid });
-        }).not.toThrowError();
+        }).not.toLogErrorDev();
+    });
+
+    // TODO [#3971]: Completely remove shadowSupportMode "any"
+    it('should warn for deprecated value "any"', () => {
+        expect(() => {
+            createElement('x-any', { is: Any });
+        }).toLogWarningDev(/Invalid value 'any' for static property shadowSupportMode/);
```
Comment (nolanlawson) [inline packages/@lwc/integration-karma/test/mixed-shadow-mode/LightningElement.shadowSupportMode/index.spec.js:26]: Can we also test that `any` renders in synthetic mode, not native mode?

File: packages/@lwc/engine-core/src/framework/def.ts [hunk 2]
```diff
@@ -130,6 +131,16 @@ function createComponentDef(Ctor: LightningElementConstructor): ComponentDef {
             );
         }
 
+        // TODO [#3971]: Completely remove shadowSupportMode "any"
+        if (
+            !isUndefined(ctorShadowSupportMode) &&
+            ctorShadowSupportMode === ShadowSupportMode.Any
```
Comment (nolanlawson) [inline packages/@lwc/engine-core/src/framework/def.ts:137]:
```suggestion
            ctorShadowSupportMode === ShadowSupportMode.Any
```

It's not necessary to test for undefined-ness here.

**Review by nolanlawson — APPROVED** [review]
Minor nits, otherwise LGTM

File: packages/@lwc/integration-karma/test/mixed-shadow-mode/LightningElement.shadowSupportMode/index.spec.js (hunk 1 above)
Comment (ekashida) [reply packages/@lwc/integration-karma/test/mixed-shadow-mode/LightningElement.shadowSupportMode/index.spec.js:26]: Sorry, I'm not sure I understand. This test runs for both synthetic and native right?

Comment (nolanlawson) [reply packages/@lwc/integration-karma/test/mixed-shadow-mode/LightningElement.shadowSupportMode/index.spec.js:26]: Right, sorry. I just mean that we should confirm that `'any'` does _not_ do the same thing as `'native'`. It should be a no-op.

**Pull request merged**